great_expectations/expectations/metrics/map_metric_provider.py

```diff
@@ -2514,17 +2503,7 @@ def _spark_column_pair_map_condition_values(
                 message=f'Error: The column "{column_name}" in BatchData does not exist.'
             )
 
-    data = (
-        df.withColumn("__row_number", F.row_number().over(Window.orderBy(F.lit(1))))
-        .withColumn("__unexpected", boolean_mapped_unexpected_values)
-        .orderBy(F.col("__row_number"))
-    )
-
-    filtered = (
-        data.filter(F.col("__unexpected") == True)
-        .drop(F.col("__unexpected"))
-        .drop(F.col("__row_number"))
-    )
+    filtered = df.filter(boolean_mapped_unexpected_values)
```

Comment:
@NathanFarmer Can we please preserve the pattern for having `data = df.withColumn("__unexpected", boolean_mapped_unexpected_values)` first -- and then followed by the filter on `F.col("__unexpected")` -- for consistency and readability purposes (plus for potential extensibility needs).  Thank you!

Reply:
See my other comments about that pattern being measurably slower. We can discuss the tradeoffs in our face-to-face testing discussion.